### PR TITLE
Add super and progressmanager to inspections

### PR DESCRIPTION
**Pull request by dlsniper**

### CODE REVIEW

File: src/com/goide/inspections/GoCgoInTestInspection.java [hunk 1]
```diff
@@ -21,11 +21,14 @@
...
 import com.goide.runconfig.testing.GoTestFinder;
 import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.progress.ProgressManager;
 import org.jetbrains.annotations.NotNull;
 
 public class GoCgoInTestInspection extends GoInspectionBase {
   @Override
   protected void checkFile(@NotNull GoFile file, @NotNull ProblemsHolder problemsHolder) {
+    super.checkFile(file, problemsHolder);
+    ProgressManager.checkCanceled();
```
Comment (zolotov) [inline src/com/goide/inspections/GoCgoInTestInspection.java]: checkCancelled is called for every element via `com.intellij.psi.PsiElementVisitor#visitElement`, so it doesn't make sense to invoke it just after `super.visit*`.

File: src/com/goide/inspections/GoDirectAssignToStructFieldInMapInspection.java [hunk 2]
```diff
@@ -33,6 +34,7 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
       public void visitLeftHandExprList(@NotNull GoLeftHandExprList o) {
         super.visitLeftHandExprList(o);
         for (GoExpression expression : o.getExpressionList()) {
+          ProgressManager.checkCanceled();
```
Comment (zolotov) [inline src/com/goide/inspections/GoDirectAssignToStructFieldInMapInspection.java]: also good

File: src/com/goide/inspections/GoCgoInTestInspection.java (hunk 1 above)
Comment (dlsniper) [inline src/com/goide/inspections/GoCgoInTestInspection.java]: Oh, ok, I've missed that. If there's a guideline for how to add this I'd be happy to update this to be more useful.

Comment (zolotov) [inline src/com/goide/inspections/GoCgoInTestInspection.java]: unfortunately there isn't, the only guideline is to invoke it on a possibly long EDT-blocking operation, especially if it runs in loop.

**Comment by dlsniper** [comment]
@zolotov I've removed most of the `checkCanceled()` calls and I've left some of them that were in for loops and such. There might be still places where it would make sense to have that but I hope it's a good start.

File: src/com/goide/inspections/GoDuplicateReturnArgumentInspection.java [hunk 3]
```diff
@@ -29,6 +30,7 @@
 public class GoDuplicateReturnArgumentInspection extends GoDuplicateArgumentInspection {
   @Override
   public void check(@Nullable GoSignature o, @NotNull ProblemsHolder holder) {
+    super.check(o, holder);
```
Comment (zolotov) [inline src/com/goide/inspections/GoDuplicateReturnArgumentInspection.java]: are you sure? looks like o.getParamters will be checked twice and also they will be check even if `GoDuplicateArgumentInspection` is disabled.

File: src/com/goide/inspections/GoAssignmentNilWithoutExplicitTypeInspection.java [hunk 4]
```diff
@@ -36,18 +37,21 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
     return new GoVisitor() {
       @Override
       public void visitVarDeclaration(@NotNull GoVarDeclaration o) {
+        super.visitVarDeclaration(o);
```
Comment (zolotov) [inline src/com/goide/inspections/GoAssignmentNilWithoutExplicitTypeInspection.java]: redundant, checkExpression invokes checkCancelled even more frequently

File: src/com/goide/inspections/GoAssignmentNilWithoutExplicitTypeInspection.java [hunk 5]
```diff
@@ -36,18 +37,21 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
...
         for (GoVarSpec spec : o.getVarSpecList()) {
           checkVar(spec);
         }
       }
 
       @Override
       public void visitShortVarDeclaration(@NotNull GoShortVarDeclaration o) {
+        super.visitShortVarDeclaration(o);
```
Comment (zolotov) [inline src/com/goide/inspections/GoAssignmentNilWithoutExplicitTypeInspection.java]: redundant, checkExpression invokes checkCancelled even more frequently

File: src/com/goide/inspections/GoAssignmentNilWithoutExplicitTypeInspection.java [hunk 6]
```diff
@@ -36,18 +37,21 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
...
         checkVar(o);
       }
 
       @Override
       public void visitConstDeclaration(@NotNull GoConstDeclaration o) {
+        super.visitConstDeclaration(o);
```
Comment (zolotov) [inline src/com/goide/inspections/GoAssignmentNilWithoutExplicitTypeInspection.java]: redundant, checkExpression invokes checkCancelled even more frequently

File: src/com/goide/inspections/GoCgoInTestInspection.java [hunk 7]
```diff
@@ -26,6 +26,7 @@
 public class GoCgoInTestInspection extends GoInspectionBase {
   @Override
   protected void checkFile(@NotNull GoFile file, @NotNull ProblemsHolder problemsHolder) {
+    super.checkFile(file, problemsHolder);
```
Comment (zolotov) [inline src/com/goide/inspections/GoCgoInTestInspection.java]: `super.checkFile` does nothing

File: src/com/goide/inspections/GoInvalidPackageImportInspection.java [hunk 8]
```diff
@@ -53,6 +54,7 @@
 
   @Override
   protected void checkFile(@NotNull GoFile file, @NotNull ProblemsHolder problemsHolder) {
+    super.checkFile(file, problemsHolder);
```
Comment (zolotov) [inline src/com/goide/inspections/GoInvalidPackageImportInspection.java]: `super.checkFile` does nothing

File: src/com/goide/inspections/GoMultiplePackagesInspection.java [hunk 9]
```diff
@@ -33,6 +33,7 @@
 public class GoMultiplePackagesInspection extends GoInspectionBase {
   @Override
   protected void checkFile(@NotNull GoFile file, @NotNull ProblemsHolder problemsHolder) {
+    super.checkFile(file, problemsHolder);
```
Comment (zolotov) [inline src/com/goide/inspections/GoMultiplePackagesInspection.java]: `super.checkFile` does nothing

File: src/com/goide/inspections/GoStructTagInspection.java [hunk 10]
```diff
@@ -37,7 +38,9 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder,
     return new GoVisitor() {
       @Override
       public void visitStructType(@NotNull GoStructType o) {
+        super.visitStructType(o);
```
Comment (zolotov) [inline src/com/goide/inspections/GoStructTagInspection.java]: redundant, you invoke checkCanceled just after this statement

File: src/com/goide/inspections/GoNoNewVariablesInspection.java [hunk 11]
```diff
@@ -41,16 +41,19 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
...
 
       @Override
       public void visitRecvStatement(@NotNull GoRecvStatement o) {
+        super.visitRecvStatement(o);
         visitVarDefinitionList(o, o.getVarDefinitionList());
       }
 
       @Override
       public void visitRangeClause(@NotNull GoRangeClause o) {
+        super.visitRangeClause(o);
```
Comment (zolotov) [inline src/com/goide/inspections/GoNoNewVariablesInspection.java]: it's easier to invoke checkCanceld in visitVarDefinitionList and remove all supers

File: src/com/goide/inspections/GoMissingReturnInspection.java [hunk 12]
```diff
@@ -134,11 +137,13 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
     return new GoVisitor() {
       @Override
       public void visitFunctionOrMethodDeclaration(@NotNull GoFunctionOrMethodDeclaration o) { // todo: extract common interface
+        super.visitFunctionOrMethodDeclaration(o);
```
Comment (zolotov) [inline src/com/goide/inspections/GoMissingReturnInspection.java]: `check` does checkCanceled

**Pull request closed without merging**